processors/pybossa_indications/processor.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from itertools import groupby
+
+import os
+import time
+import requests
+import pbclient as pbc
+import logging
+logger = logging.getLogger(__name__)
+
+PYBOSSA_URL = os.environ['PYBOSSA_URL']
+PYBOSSA_API_KEY = os.environ['PYBOSSA_API_KEY']
+PROJECT_ID = os.environ['PYBOSSA_PROJECT_INDICATIONS']
+
+
+def process(conf, conn):
+
+    # Build up a limit so we'l get all the tasks in one go
+    offset = 0
+    bufsize = 10000
+
+    # Get stuff from the DB
+    query = """
+        SELECT
+            documents.id as meta_id,
+            fda_approvals.id,
+            documents.name,
+            documents.url,
+            fda_approvals.type,
+            fda_approvals.notes,
+            fda_approvals.supplement_number
+        FROM documents
+        INNER JOIN fda_approvals ON documents.fda_approval_id = fda_approvals.id
+        WHERE documents.url IS NOT NULL
+        AND fda_approvals.type = 'New or Modified Indication'
+        GROUP BY fda_approvals.id, documents.id
+        ORDER BY fda_approvals.id LIMIT %s OFFSET %s
+    """
+
+    rows = list(conn['database'].query(query % (bufsize, offset)))
+    if not rows:
+        logger.info('No new and modified indications found for tasks creation')
+        return
+    create_tasks(rows)
+
+
+def create_tasks(rows):
+    tasks = []
+    for key, group in groupby(rows, lambda x: x['id']):
+        task = {}
+        task['documents'] = []
+
+        for doc in group:
+            task['supplement_number'] = str(doc['supplement_number'])
+            task['approval_type'] = str(doc['type'])
+            task['approval_id'] = str(doc['id'])
+            task['meta_id'] = str(doc['meta_id'])
+            task['notes'] = doc['notes'] or ""
+            task['documents'].append({
+                'name': doc['name'],
+                'url': doc['url']
+            })
```

Comment:
You can change this for:

```python
task = {
  'supplement_number': str(doc[0]['supplement_number']),
  # ...
}

task['documents'] = [{'name': doc['name'], 'url': doc['url']}
                     for doc in group]

tasks.append(task)
```

Also, instead of `doc['notes'] or ""` use `doc.get('notes', '')`

Do you need to cast everything to `str()`? Just asking because all fields except `supplement_number` and maybe `id` should be strings already.